adept/_vlasov1d/storage.py

```diff
@@ -136,21 +132,107 @@ def fields_save_func(t, y, args):
     return fields_save_func
 
 
-def get_dist_save_func(cfg, k):
-    if {"t"} == set(cfg["save"][k].keys()):
+def get_dist_save_func(axes, dist_save_config, dist_key):
+    if {"t"} == set(dist_save_config.keys()):
 
         def dist_save_func(t, y, args):
-            return y["electron"]
+            return y[dist_key]
+
+    elif {"t", "x", "v"} == set(dist_save_config.keys()):
+
+        def dist_save_func(t, y, args):
+            fkx = jnp.fft.rfft(y[dist_key], axes=0)
+            return interp2d(
+                dist_save_config["x"]["ax"],
+                dist_save_config["v"]["ax"],
+                axes["x"],
+                axes["v"],
+                fkx,
+                kind="linear",
+            )
+
+    elif {"t", "kx", "v"} == set(dist_save_config.keys()):
 
-    elif {"t", "x", "y"} == set(cfg["save"][k].keys()):
+        def dist_save_func(t, y, args):
+            fkx = jnp.fft.rfft(y[dist_key], axes=0)
+            return interp2d(
+                dist_save_config["kx"]["ax"],
+                dist_save_config["v"]["ax"],
+                axes["kx"],
+                axes["v"],
+                fkx,
+                kind="linear",
+            )
+
+    elif {"t", "x", "kv"} == set(dist_save_config.keys()):
         pass
-    elif {"t", "kx", "y"} == set(cfg["save"][k].keys()):
+
+    elif {"t", "kx", "kv"} == set(dist_save_config.keys()):
         pass
+    else:
+        raise NotImplementedError
 
-    elif {"t", "x", "ky"} == set(cfg["save"][k].keys()):
+    return dist_save_func
+
+
+def get_diag_save_func(axes, dist_save_config, dist_key):
+    if {"t"} == set(dist_save_config.keys()):
+
+        def dist_save_func(t, y, args):
+            return {"diag-vlasov": y["diag-vlasov"], "diag-fp": y["diag-fp"]}
```

Comment:
same here, could do `{diag_name: y[diag_name] for diag_name in cfg["diagnostics"]}`

Reply:
combined this with `get_dist_save_func()`